examples/cotengra_setting_bench.py

```diff
@@ -0,0 +1,141 @@
+"""
+Optimization for performance comparison with different cotengra settings.
+"""
+
+import itertools
+import sys
+import warnings
+
+import cotengra as ctg
+import networkx as nx
+import numpy as np
+
+sys.path.insert(0, "../")
+import tensorcircuit as tc
+
+try:
+    import kahypar
+except ImportError:
+    print("kahypar not installed, please install it to run this script.")
+    exit()
+
+
+# suppress the warning from cotengra
+warnings.filterwarnings(
+    "ignore",
+    message="The inputs or output of this tree are not ordered."
+    "Costs will be accurate but actually contracting requires "
+    "ordered indices corresponding to array axes.",
+)
+
+K = tc.set_backend("jax")
+
+
+def generate_circuit(param, g, n, nlayers):
+    # construct the circuit ansatz
+    c = tc.Circuit(n)
+    for i in range(n):
+        c.H(i)
+    for j in range(nlayers):
+        c = tc.templates.blocks.QAOA_block(c, g, param[j, 0], param[j, 1])
+    return c
+
+
+def trigger_cotengra_optimization(n, nlayers, d):
+    g = nx.random_regular_graph(d, n)
+
+    # define the loss function
+    def loss_f(params, n, nlayers):
+
+        c = generate_circuit(params, g, n, nlayers)
+
+        # calculate the loss function, max cut
+        loss = 0.0
+        for e in g.edges:
+            loss += c.expectation_ps(z=[e[0], e[1]])
+
+        return K.real(loss)
+
+    params = K.implicit_randn(shape=[nlayers, 2])
+
+    # run only once to trigger the compilation
+    K.jit(
```

Comment:
why using jit and grad here?

Reply:
Actually, I find that jitting the function here speeds up the path-finding time. As you pointed out, the gradient calculation was unnecessary, so I removed it.